cmake/FindSYCL.cmake

```diff
@@ -0,0 +1,193 @@
+# This file is part of the Acts project.
+#
+# Copyright (C) 2020 CERN for the benefit of the Acts project
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This module is used to find out whether the C++ compiler set up to build the
+# project is able to compile SYCL code. If so, it provides helper variables
+# to the project configuration to set up the build of SYCL
+# libraries/executables.
+#
+# Variable(s) set up by the module:
+#  - SYCL_FOUND: Boolean, set to TRUE when SYCL compilation is available.
+#  - acts_setup_sycl_target( <target> [DEPENDENCY PUBLIC|PRIVATE|INTERFACE] ):
```

Comment:
As far as I understand it, this CMake module is not Acts specific? For consistency with other CMake command, the helper function should probably be called `target_setup_sycl` or `acts_target_setup_sycl` if you want to retain the `acts_` prefix. This matches e.g. `target_link_libraries(...)`.

Reply:
Hi Moritz,

The naming here is consistent with the naming that I chose in the ATLAS offline code.

https://gitlab.cern.ch/atlas/atlasexternals/-/blob/master/Build/AtlasCMake/modules/FindSYCL.cmake

I see your point, but it's not really a strong one. This function is not all that analogous to [target_link_libraries(...)](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) that we should use that as a template for its name.

Since we were the ones who wrote this code, the `acts_` prefix seems appropriate to me. Intel did not produce any convenient CMake integration for their compiler just yet...

Cheers,
        Attila

Reply:
I chose the later option. :)

(p.s. I hope you don't think I was able to write this CMake file.)

Reply:
@krasznaa Thanks for the clarification. My line of though was that it modifies an existing target similar to what most of the `target_...` functions do. I am Ok either way.